app/models/account.rb

```diff
@@ -0,0 +1,5 @@
+class Account < ApplicationRecord
+  belongs_to :owner, class_name: "User", foreign_key: "user_id", optional: true
```

Comment:
change the foreign_key to be owner (you will need to change this in the migration as well), and the owner should not be optional.

Reply:
We may add dynamic it is a new record, because if it is removed completely raise a validation error

```
> user.account.errors
 => #<ActiveModel::Errors [#<ActiveModel::Error attribute=owner, type=blank, options={:message=>:required}>]> 
> user.account.errors.full_messages
 => ["Owner must exist"] 
 ```
 
 I want to use:
 ```
 class Account < ApplicationRecord
   belongs_to :owner, class_name: "User", foreign_key: "owner_id", inverse_of: :account, optional: -> { new_record? }
   ...
  ```
  I am open to other suggestions, thanks! :)

Reply:
```ruby
class Account < ApplicationRecord
  belongs_to :owner, class_name: "User", inverse_of: :account
end

class User < ApplicationRecord
  has_one :account, foreign_key: "owner_id", inverse_of: :owner
end
```

I think this should work.

Reply:
I like your solution, more clear 👍 :)